src/logic/calculate.js

```diff
@@ -0,0 +1,28 @@
+/* eslint-disable no-return-assign */
+/* eslint-disable consistent-return */
+import operate from './operate';
+
+const calculate = (calData, buttonName) => {
+  let { total, next, operation } = calData;
+
+  switch (buttonName) {
+    case '+/-':
+      total = -total;
+      next = -next;
+      operation = buttonName;
+      break;
+    case 'AC':
+      total = 0;
+      break;
+    case '.':
+      if (next) {
+        return total = [total, '.', next].join('');
+      }
+      break;
+    default:
+      return operate(total, next, operation);
+  }
+};
```

Comment:
The `calculate` function seems not to be returning anything for cases when `buttonName` is '+/-' or 'AC'.

Why not create an object to be returned at the end of the `switch...case` that would include the new state (total, next and operator)?